src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -51,6 +53,23 @@ internal ExceptionDataVersionOne CreateExceptionData(ExceptionData v2ExceptionDa
             return exceptionData;
         }
 
+        internal FileChangeVersionOne CreateFileChange(FileChange v2FileChange)
+        {
+            FileChangeVersionOne fileChange = null;
+
+            if (v2FileChange != null)
+            {
+                fileChange = new FileChangeVersionOne
+                {
+                    Replacements = v2FileChange.Replacements?.Select(r => CreateReplacement(r, v2FileChange.FileLocation?.Uri)).ToList(),
```

Comment:
As the code stands, you would recompute the encoding for each replacement, even though the encoding is the same for all replacements in the file. This suggests that `CreateReplacement` isn't logically the right place for that computation. Instead, I'd do the following:

- Introduce a helper that implements the logic I described in a previous comment. Note that it swallows the exception for an unrecognized encoding and just returns `null`, because we don't yet know if we will _need_ the encoding.
```
private static Encoding GetTextEncoding(Uri uri)
{
    Encoding encoding = null;

    FileData fileData;
    if (uri != null
        && currentV2Run.Files.TryGetValue(uri.OriginalString, out fileData)
        && fileData.Encoding != null)
    {
        try {
            encoding = Encoding.GetEncoding(fileData.Encoding);
        }
        catch (ArgumentException)
        {
        }
    }

    return encoding
}
```
- Call `GetTextEncoding` from at the top of `CreateFileChange`:
```
    Encoding encoding = GetTextEncoding(v2FileChange.FileLocation?.Uri);
```
- Pass the _encoding_, not the URI, to `CreateReplacement` (note that that's the only thing `CreateReplacement` actually needs):
```
internal ReplacementVersionOne CreateReplacement(v2Replacement, Encoding encoding)
{
    ...
}
```
- Within `CreateReplacement`, if you find that you need the encoding but it's `null`, _then_ throw the `UnknownEncodingException`.

Reply:
Much better. The only sacrifice is that the unknown encoding name is only available within the helper method.

Reply:
Have a refactor that should fix this...